core/src/main/java/org/apache/metamodel/query/SelectItem.java

```diff
@@ -461,8 +472,11 @@ public StringBuilder toStringNoAlias() {
     public StringBuilder toStringNoAlias(boolean includeSchemaInColumnPath) {
         final StringBuilder sb = new StringBuilder();
         if (_column != null) {
-            sb.append(getToStringColumnPrefix(includeSchemaInColumnPath));
-            sb.append(_column.getQuotedName());
+            if (_function==null ||
+                    (_function!=null && !_function.getFunctionName().equals(FunctionType.CONCAT.getFunctionName()))) {
```

Comment:
Unclear to me why this is needed. I would prefer this to be really agnostic of function type. Any function should be able to have a mix of parameter types (if that is somehow the reason for this if - I'm unsure).